kedro_neptune/__init__.py

```diff
@@ -80,99 +107,130 @@ def init(metadata: ProjectMetadata, api_token: str, project: str, base_namespace
+        return run[base_namespace]
 
-        return self._run[self._base_namespace]
 
+class BinaryFileDataSet(TextDataSet):
+    def __init__(
+            self,
+            filepath: str,
+            version: Version = None,
+            credentials: Dict[str, Any] = None,
+            fs_args: Dict[str, Any] = None,
+    ) -> None:
+        super(BinaryFileDataSet, self).__init__(
+            filepath=filepath,
+            version=None,
+            credentials=credentials,
+            fs_args=fs_args
+        )
 
-class AbstractNeptuneDataSet(AbstractDataSet, ABC):
-    """Abstract class for extending other DataSets.
-     Instances are produced by NeptuneArtifactDataSet factory."""
+        self._fs_open_args_load.setdefault("mode", "rb")
+        self._fs_open_args_save.setdefault("mode", "wb")
 
-    def load_raw(self) -> Any:
-        pass
+    def _describe(self) -> Dict[str, Any]:
+        load_path = get_filepath_str(self._get_load_path(), self._protocol)
 
+        path = urllib.parse.urlparse(load_path).path
+        extension = os.path.splitext(path)[1][1:]
 
-class NeptuneArtifactDataSet(AbstractDataSet):
-    def __new__(cls, dataset: Dict):
-        dataset_class, data_set_args = parse_dataset_definition(config=dataset)
+        return dict(
+            extension=extension,
+            **super(BinaryFileDataSet, self)._describe()
+        )
 
-        class NeptuneExtendedDataSet(dataset_class, AbstractNeptuneDataSet):
-            """This class extends dataset_class.
-            It's kind of 'annotation' with information that this `dataset` should be uploaded to neptune."""
-            def __init__(self):
-                super().__init__(**data_set_args)
+    def _save(self, data: bytes) -> None:
+        load_path = get_filepath_str(self._get_load_path(), self._protocol)
 
-            def load_raw(self):
-                load_path = get_filepath_str(self._get_load_path(), self._protocol)
+        with self._fs.open(load_path, **self._fs_open_args_load) as fs_file:
+            return fs_file.read()
 
-                with self._fs.open(load_path, **self._fs_open_args_load) as fs_file:
-                    return fs_file.read()
+    def _load(self) -> bytes:
+        load_path = get_filepath_str(self._get_load_path(), self._protocol)
 
-        # rename the class
-        parent_name = dataset_class.__name__
-        NeptuneExtendedDataSet.__name__ = f"NeptuneArtifactDataSetFor{parent_name}"
-        NeptuneExtendedDataSet.__qualname__ = f"{cls.__name__}.{NeptuneArtifactDataSet.__name__}"
+        with self._fs.open(load_path, **self._fs_open_args_load) as fs_file:
+            return fs_file.read()
 
-        # pylint: disable=abstract-class-instantiated
-        return NeptuneExtendedDataSet()
 
-    def _load(self) -> Any:
-        raise AttributeError("NeptuneArtifactDataSet is a factory, this function should not be called.")
+class NeptuneFileDataSet(BinaryFileDataSet):
```

Comment:
can you explain the need for both `NeptuneFileDataSet` and `BinaryFileDataSet` existing? :)

Reply:
I think it should be included in Kedro datasets along with TextFileDataSet (in future). It's only about distinct domains and it will be confusing that NeptuneFileDataSet inherits from TestFileDataSet but uploading as binary.